R/backend-teradata.R

```diff
@@ -109,8 +109,11 @@ sql_translation.Teradata <- function(con) {
                         digits <- vctrs::vec_cast(digits, integer())
                         sql_expr(CAST(!!x %as% DECIMAL(12L, !!digits)))
                       },
-      as.double     = sql_cast("NUMERIC"),
-      as.character  = sql_cast("VARCHAR(MAX)"),
+      as.double     = sql_cast("FLOAT"),
+      as.character  = function(x, nchar = 255L) {
+        nchar <- vctrs::vec_cast(nchar, integer())
```

Comment:
```suggestion
        check_number_whole(nchar, min = 0, max = 255)
```

Yup, we do have the type checkers! This change should do the trick, @rplsmn—once you've accepted it and checks pass, this PR will be good to go.

Reply:
[Reference for that 64,000 value](https://www.sfu.ca/sasdoc/sashtml/accdb/z1250040.htm#:~:text=character%20string%20data.-,The%20maximum%20for%20n%20is%2064%2C000.,also%20known%20as%20CHARACTER%20VARYING.&text=specifies%20a%20varying%2Dlength%20column%2C%20of%20the%20maximum%20length%2C,version%20your%20server%20is%20running.).